modules/vba_documents/lib/vba_documents/upload_validator.rb

```diff
@@ -52,7 +52,7 @@ def validate_metadata(metadata_input, consumer_id, upload_guid, submission_versi
       if rejected.present?
         raise VBADocuments::UploadError.new(code: 'DOC102', detail: "Non-string values for keys: #{rejected.join(',')}")
       end
-      if (FILE_NUMBER_REGEX =~ metadata['fileNumber']).nil?
+      if (FILE_NUMBER_REGEX =~ metadata['fileNumber']&.strip).nil?
```

Comment:
The safe navigation operator (`&.`) isn't needed here because L48 will raise an exception if there are missing keys, and L53 will raise an exception if the value isn't a string. Therefore `metadata['fileNumber']` won't ever be `nil` if L55 is reached.

Reply:
fixedremoved